lib/Service/ImageService.php

```diff
@@ -0,0 +1,658 @@
+			$ownerParentFolder = $ownerTextFile->getParent();
+			$attachmentFolderName = '.attachments.' . $textFile->getId();
+			if ($ownerParentFolder->nodeExists($attachmentFolderName)) {
+				$attachmentFolder = $ownerParentFolder->get($attachmentFolderName);
+				if ($attachmentFolder instanceof Folder) {
+					return $attachmentFolder;
+				}
+			} elseif ($create) {
+				return $ownerParentFolder->newFolder($attachmentFolderName);
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Get a user file from file ID
+	 * @param string $filePath
+	 * @param string $userId
+	 * @return File|null
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	private function getFileFromPath(string $filePath, string $userId): ?File {
+		$userFolder = $this->rootFolder->getUserFolder($userId);
+		if ($userFolder->nodeExists($filePath)) {
+			$file = $userFolder->get($filePath);
+			if ($file instanceof File) {
+				return $file;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Get a user file from file ID
+	 *
+	 * @param int $documentId
+	 * @param string $userIdd
+	 * @return File
+	 * @throws NotFoundException
+	 * @throws NotPermittedException
+	 * @throws \OC\User\NoUserException
+	 */
+	private function getTextFile(int $documentId, string $userId): File {
+		$userFolder = $this->rootFolder->getUserFolder($userId);
+		$textFile = $userFolder->getById($documentId);
+		if (count($textFile) > 0 && $textFile[0] instanceof File) {
+			return $textFile[0];
+		} else {
+			throw new NotFoundException('Text file with id=' . $documentId . ' was not found in storage of ' . $userId);
+		}
+	}
+
+	/**
+	 * Get file from share token
+	 *
+	 * @param int|null $documentId
+	 * @param string $shareToken
+	 * @return File
+	 * @throws NotFoundException
+	 */
+	private function getTextFilePublic(?int $documentId, string $shareToken): File {
+		// is the file shared with this token?
+		try {
+			$share = $this->shareManager->getShareByToken($shareToken);
+			if ($share->getShareType() === IShare::TYPE_LINK) {
+				// shared file or folder?
+				if ($share->getNodeType() === 'file') {
+					$textFile = $share->getNode();
+					if ($textFile instanceof File) {
+						return $textFile;
+					}
+				} elseif ($documentId !== null && $share->getNodeType() === 'folder') {
+					$folder = $share->getNode();
+					if ($folder instanceof Folder) {
+						$textFile = $folder->getById($documentId);
+						if (count($textFile) > 0 && $textFile[0] instanceof File) {
+							return $textFile[0];
+						}
+					}
+				}
+			}
+		} catch (ShareNotFound $e) {
+			// same as below
+		}
+		throw new NotFoundException('Text file with id=' . $documentId . ' and shareToken ' . $shareToken . ' was not found.');
+	}
+
+	/**
+	 * Download a file and write it to a resource
+	 * @param string $url
+	 * @param $resource
+	 * @param array $params
+	 * @return array
+	 */
+	private function simpleDownload(string $url, $resource, array $params = []): array {
```

Comment:
The parameter `$params` is never passed to the private function `simpleDownload()` and probably can be dropped, no?

Reply:
:+1: